src/librbd/io/ImageDispatchSpec.cc

```diff
@@ -131,6 +131,20 @@ uint64_t ImageDispatchSpec<I>::extents_length() {
   return length;
 }
 
+template <typename I>
+std::pair<uint64_t, uint64_t> ImageDispatchSpec<I>::get_image_extents() {
+   std::pair<uint64_t, uint64_t> image_extents = {0, 0};
+   if(this->m_image_extents.size()) {
+     image_extents = this->m_image_extents.front();
```

Comment:
Nit: just return the `front()` here ... and avoid pre-initializing the worst-case above.

Reply:
Done

Reply:
Yes, understood. Just return `front()` directly (since you have already verified it has at least a single entry). The flush case can directly return the `{0, 0}` pair.

Reply:
> The flush case can directly return the `{0, 0}` pair.

This is accessed in `process()` which means I would need a mechanism to identify the flush operation. Could you explain why such an initialization is bad, particularly here? I suppose it's fine to initialize elsewhere where this function is called?

Reply:
The flush case doesn't matter, you are really just saying that it has no IO extents. The fact that you are pre-initializing a `{0, 0}` pair for all cases and then only using the result for flushes is somewhat arbitrary. Passing back the full image extents would avoid this since then block/unblock would see that there are no associated extents.

Reply:
Done. It now returns full extents vector.